libcontainer/seccomp/seccomp_linux.go

```diff
@@ -33,54 +34,85 @@ const (
 // Started in the container init process, and carried over to all child processes
 // Setns calls, however, require a separate invocation, as they are not children
 // of the init until they join the namespace
-func InitSeccomp(config *configs.Seccomp) error {
+func InitSeccomp(config *configs.Seccomp) (int, error) {
 	if config == nil {
-		return errors.New("cannot initialize Seccomp - nil config passed")
+		return -1, errors.New("cannot initialize Seccomp - nil config passed")
 	}
 
 	defaultAction, err := getAction(config.DefaultAction, nil)
 	if err != nil {
-		return errors.New("error initializing seccomp - invalid default action")
+		return -1, errors.New("error initializing seccomp - invalid default action")
 	}
 
 	filter, err := libseccomp.NewFilter(defaultAction)
 	if err != nil {
-		return fmt.Errorf("error creating filter: %s", err)
+		return -1, fmt.Errorf("error creating filter: %s", err)
+	}
+
+	notifyFeatureRequired := false
+	for _, call := range config.Syscalls {
+		if call.Action == configs.Notify {
+			notifyFeatureRequired = true
+			break
+		}
+	}
```

Comment:
Add a TODO in the code:
```
// TODO: check that it is consistent with config.Flags
```

runtime-spec defines `type LinuxSeccompFlag string` but it does not seem to be used. We should at least check that if we have `notifyFeatureRequired`, then we have `SECCOMP_FILTER_FLAG_NEW_LISTENER` and possibly `SECCOMP_FILTER_FLAG_TSYNC` and `SECCOMP_FILTER_FLAG_TSYNC_ESRCH`...

And if `notifyFeatureRequired` is not set, then `SECCOMP_FILTER_FLAG_NEW_LISTENER` must not be present.

Reply:
I added a similar comment and I'll take a closer look once the runc-spec PR is ready to merge.